diff --git a/registry_rebuild.drush.inc b/registry_rebuild.drush.inc
index 8bab938..8d3c29c 100644
--- a/registry_rebuild.drush.inc
+++ b/registry_rebuild.drush.inc
@@ -58,6 +58,7 @@ function drush_registry_rebuild() {
   $includes = array(
     $include_dir . '/bootstrap.inc',
     $include_dir . '/common.inc',
+    $include_dir . '/entity.inc',
     $module_dir . '/entity/entity.module',
     $module_dir . '/entity/entity.controller.inc',
     $module_dir . '/system/system.module',
